src/transformers/modeling_utils.py

```diff
@@ -638,15 +645,7 @@ def invert_attention_mask(self, encoder_attention_mask: Tensor) -> Tensor:
         # encoder_extended_attention_mask = (encoder_extended_attention_mask ==
         # encoder_extended_attention_mask.transpose(-1, -2))
         encoder_extended_attention_mask = encoder_extended_attention_mask.to(dtype=self.dtype)  # fp16 compatibility
-
-        if self.dtype == torch.float16:
-            encoder_extended_attention_mask = (1.0 - encoder_extended_attention_mask) * -1e4
-        elif self.dtype in [torch.bfloat16, torch.float32]:
-            encoder_extended_attention_mask = (1.0 - encoder_extended_attention_mask) * -1e9
-        else:
-            raise ValueError(
-                f"{self.dtype} not recognized. `dtype` should be set to either `torch.float32` or `torch.float16`"
-            )
+        encoder_extended_attention_mask = (1.0 - encoder_extended_attention_mask) * mask_value(self.dtype, self.device)
```

Comment:
`self.device` shouldn't be used here as it breaks model parallelism. If we accept the changes to `mask_value` above, this will be a float and there won't be any need to put it on any device manually.
```suggestion
        encoder_extended_attention_mask = (1.0 - encoder_extended_attention_mask) * mask_value(self.dtype)
```